### PR TITLE
IDEX-2905 Allow to use Docker Machine system variables (master branch)

### PR DESCRIPTION
**Pull request by benoitf**
Variables define the boot2docker configuration (like ip address, protocol, certificates folder, etc)

### CODE REVIEW

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java [hunk 1]
```diff
@@ -47,11 +63,57 @@ public DockerConnectorConfiguration(URI dockerDaemonUri,
...
+                // unable to use given property, fallback to default URL
+                return DEFAULT_BOOT2DOCKER_URI;
+
+            }
+        }
+
+
+
+
+
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: Generally it is okay, but come on is it really necessary to have 5 empty lines here?

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java [hunk 2]
```diff
@@ -14,14 +14,30 @@
...
+import static org.eclipse.che.plugin.docker.client.DockerConnector.DEFAULT_BOOT2DOCKER_URI;
+import static org.eclipse.che.plugin.docker.client.DockerConnector.UNIX_SOCKET_URI;
 
 /**
  * @author Alexander Garagatyi
  */
 public class DockerConnectorConfiguration {
+
+    /**
+     * Logger.
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: We are always welcome for javadoc, but what the sense of this one?

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java [hunk 3]
```diff
@@ -14,14 +14,30 @@
...
+
+import static org.eclipse.che.plugin.docker.client.DockerConnector.BOOT2DOCKER_CERT_PATH_PROPERTY;
+import static org.eclipse.che.plugin.docker.client.DockerConnector.BOOT2DOCKER_HOST_PROPERTY;
+import static org.eclipse.che.plugin.docker.client.DockerConnector.BOOT2DOCKER_TLS_VERIFY_PROPERTY;
+import static org.eclipse.che.plugin.docker.client.DockerConnector.DEFAULT_BOOT2DOCKER_CERTS_DIR;
+import static org.eclipse.che.plugin.docker.client.DockerConnector.DEFAULT_BOOT2DOCKER_URI;
+import static org.eclipse.che.plugin.docker.client.DockerConnector.UNIX_SOCKET_URI;
 
 /**
  * @author Alexander Garagatyi
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: Can you please add yourself as a <code>@author</code>

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 2 above)
Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: well I comment all what I add in general. Looks more pretty. Checkstyle or sonar should be happier as well. I should remove it ?

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 1 above)
Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: oops. Will fix it asap

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 3 above)
Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: ok

Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: +1

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 2 above)
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: yes

Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: yes I should remove ? not a joke ?

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 3 above)
Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: done

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 1 above)
Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: done

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 2 above)
Comment (skabashnyuk) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: Up to you. You can leave it as is. But it's a dummy comment - no?

Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: OK, I prefer to leave it as this. Just habit to have everything commented on top of method and inside methods describing logic

**Comment by skabashnyuk** [comment]
ok

**Comment by benoitf** [comment]
Rethinking about it, I may add some tests but I would require to mock System.getEnv() (^^)
So instead of trying to mock java.lang, I would need to add a new method that carry on the Environment. Would it be OK then ?

something like   private static URI dockerDaemonUri() 
    redirecting to dockerDaemonUri(System.getEnv()))

and another method dockerDaemonUri that will take Map<String, String>

**Comment by benoitf** [comment]
Ok I've added tests

**Comment by evoevodin** [comment]
It seems everything is okay, well done :+1:

**Comment by benoitf** [comment]
Also I add PR for 4.0 branch
https://github.com/codenvy/che-plugins/pull/206

**Comment by skabashnyuk** [comment]
ok go

File: plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java (hunk 2 above)
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: I also dislike useless and obvious comments. And I remove such comments when I see them. Please, do not commit such things anymore. By the way see logger declaration standart https://wiki.codenvycorp.com/display/DPP/Logger

Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: If it's final and static, it should be in uppercase as well
LOGGER and not logger to be compliant.

About comments, you may dislike it but I'm conforming to code standard http://www.oracle.com/technetwork/articles/java/index-137868.html.

Comment (benoitf) [inline plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnectorConfiguration.java]: Also I don't like when interface are not commented as well
looking at https://github.com/codenvy/che-plugins/blob/4.0/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineFactory.java

**Pull request merged**